### DISCUSSION TITLE
CollectionView virtual (infinite) scrolling - how many items to download and add initially?

### DISCUSSION THREAD

**Discussion by paratype-fonts** [dotnet/maui#18279] [Q&A]
When initially populating CollectionView items, I need to know when a new item becomes not shown (clipped) to stop fetching. 

The Scrolled event has very useful properties: LastVisibleItemIndex and FirstVisibleItemIndex, but the number of items to fetch needs to be determined before scroll occurs. I could not find anything functionally similar on the CollectionView itself. 

Any suggestions? Thanks!

**Comment by dimustriz** [comment]
As a workaround, you can override MeasureOverride and match height (and width if needed) with DesiredSize props.
